### PR TITLE
More fixes

### PR DESCRIPTION
**Pull request by jschwartzenberg**

### CODE REVIEW

File: maven-core/src/main/java/org/apache/maven/lifecycle/internal/builder/multithreaded/ConcurrencyDependencyGraph.java [hunk 1]
```diff
@@ -61,20 +62,20 @@ public int getNumberOfBuilds()
...
+        Set<MavenProject> result = new LinkedHashSet<>();
         for ( ProjectSegment projectBuild : projectBuilds )
         {
             if ( projectDependencyGraph.getUpstreamProjects( projectBuild.getProject(), false ).isEmpty() )
             {
                 result.add( projectBuild.getProject() );
             }
         }
-        return result;
+        return new ArrayList<>( result );
```
Comment (dizzzz) [inline maven-core/src/main/java/org/apache/maven/lifecycle/internal/builder/multithreaded/ConcurrencyDependencyGraph.java:78]: Here there is not a set returned, but an (array) list: java doc is inconsitent

File: maven-core/src/main/java/org/apache/maven/lifecycle/internal/builder/multithreaded/MultiThreadedBuilder.java [hunk 2]
```diff
@@ -112,6 +114,9 @@ public void build( MavenSession session, ReactorContext reactorContext, ProjectB
             }
 
         }
+
+        executor.shutdown();
+        executor.awaitTermination( Long.MAX_VALUE, TimeUnit.MILLISECONDS  );
```
Comment (dizzzz) [inline maven-core/src/main/java/org/apache/maven/lifecycle/internal/builder/multithreaded/MultiThreadedBuilder.java:119]: MAXVALUE... what is the risk of hanging builds in CI? If any, lowering might be a good idea?

**Review by dizzzz — APPROVED** [review]
Small changes needed?

**Comment by jschwartzenberg** [comment]
@dizzzz I think you should offer your services to the Maven project :)

**Pull request merged**